src/systemcmds/pwm/pwm.c

```diff
@@ -649,18 +671,26 @@ pwm_main(int argc, char *argv[])
 						}
 					}
 
-					warnx("User abort\n");
+					PX4_INFO("User abort\n");
 					return 0;
 				}
 			}
 
-			usleep(2000);
+			/* Delay longer than the max Oneshot duration */
+
+			usleep(2542);
```

Comment:
We are not running tickless. 

The 25[42](https://en.wikipedia.org/wiki/42_(number)#The_Hitchhiker.27s_Guide_to_the_Galaxy) was chosen for its special properties. :) 
 Per Open group it will be rounded [see Implementations may place limitations on the granularity](http://pubs.opengroup.org/onlinepubs/009695399/functions/usleep.html) to 3 then quantized up to 4 Ms

nsh> pwm test -a -p 2010 -d /dev/pwm_output1 (2010 * 125*10^-9) = 251.25uS
![image](https://cloud.githubusercontent.com/assets/1945821/23823089/b497d24e-05fe-11e7-9393-e6c7f4a2a241.png)